services/mgmt/vmware/src/package_2021_12_01/operations.rs

```diff
@@ -200,46 +202,150 @@ pub enum Error {
+        Self {
+            endpoint,
+            credential,
+            scopes,
+            pipeline,
+        }
+    }
+
+    pub fn operations(&self) -> operations::Client {
+        operations::Client(self.clone())
+    }
+}
+
 pub mod operations {
     use super::{models, API_VERSION};
-    pub async fn list(operation_config: &crate::OperationConfig) -> std::result::Result<models::OperationList, list::Error> {
-        let http_client = operation_config.http_client();
-        let url_str = &format!("{}/providers/Microsoft.AVS/operations", operation_config.base_path(),);
-        let mut url = url::Url::parse(url_str).map_err(list::Error::ParseUrlError)?;
-        let mut req_builder = http::request::Builder::new();
-        req_builder = req_builder.method(http::Method::GET);
-        if let Some(token_credential) = operation_config.token_credential() {
-            let token_response = token_credential
-                .get_token(operation_config.token_credential_resource())
-                .await
-                .map_err(list::Error::GetTokenError)?;
-            req_builder = req_builder.header(http::header::AUTHORIZATION, format!("Bearer {}", token_response.token.secret()));
-        }
-        url.query_pairs_mut().append_pair("api-version", super::API_VERSION);
-        let req_body = bytes::Bytes::from_static(azure_core::EMPTY_BODY);
-        req_builder = req_builder.uri(url.as_str());
-        let req = req_builder.body(req_body).map_err(list::Error::BuildRequestError)?;
-        let rsp = http_client.execute_request(req).await.map_err(list::Error::ExecuteRequestError)?;
-        match rsp.status() {
-            http::StatusCode::OK => {
-                let rsp_body = rsp.body();
-                let rsp_value: models::OperationList =
-                    serde_json::from_slice(rsp_body).map_err(|source| list::Error::DeserializeError(source, rsp_body.clone()))?;
-                Ok(rsp_value)
-            }
-            status_code => {
-                let rsp_body = rsp.body();
-                let rsp_value: models::CloudError =
-                    serde_json::from_slice(rsp_body).map_err(|source| list::Error::DeserializeError(source, rsp_body.clone()))?;
-                Err(list::Error::DefaultResponse {
-                    status_code,
-                    value: rsp_value,
-                })
-            }
+
+    pub struct Client(pub(crate) super::Client);
+
+    impl Client {
+        pub fn list<'a>(&'a self, name: &'a str) -> list::Builder {
+            list::Builder { client: self.0.clone(), name }
         }
     }
+
     pub mod list {
         use super::{models, API_VERSION};
+
+        #[derive(Clone)]
+        pub struct Builder<'a> {
+            pub(crate) client: crate::operations::Client,
+            pub(crate) name: &'a str,
+        }
+
+        impl<'a> Builder<'a> {
+            pub fn into_future(self) -> futures::future::BoxFuture<'a, std::result::Result<models::OperationList, Error>> {
```

Comment:
But then the future is limited to that lifetime. Is that okay or do we want a `'static` lifetime?

Reply:
For #527, I left the parameters in `Builder` as owned so that the `BoxFuture` lifetime is `'static`.